### PR TITLE
samples: Bluetooth: df: Add DF central sample

### PR DESCRIPTION
**Pull request by ppryga-nordic**
Add new sample application for direction finding to show
how the functionality works in connected mode for central role.

Signed-off-by: Piotr Pryga <piotr.pryga@nordicsemi.no>

### CODE REVIEW

**Comment by ppryga** [comment]
The CI fails due to wrong Kconfig name that was accidentally merged in the past. The name is corrected in PR: https://github.com/zephyrproject-rtos/zephyr/pull/41905/files#diff-e9984b2a061624f67ff5705bc3295a77b97df2b1c5b4452dc706a3177bd710c1
This PR has to be rebased and merged after the #41905 is merged.

File: samples/bluetooth/direction_finding_central/boards/nrf52833dk_nrf52820.conf
Comment (Thalley) [inline samples/bluetooth/direction_finding_central/boards/nrf52833dk_nrf52820.conf:23]: Missing empty line

File: samples/bluetooth/direction_finding_central/src/main.c
Comment (Thalley) [inline samples/bluetooth/direction_finding_central/src/main.c:22]: Looks like it is set to 5, and not 0.

File: samples/bluetooth/direction_finding_central/src/main.c
Comment (Thalley) [inline samples/bluetooth/direction_finding_central/src/main.c:72]: This could perhaps be moved to `direction.h` - Possible just as `inline` functions. It isn't something we've done a lot in the past, but possible a useful feature to have.

File: samples/bluetooth/direction_finding_central/src/main.c
Comment (Thalley) [inline samples/bluetooth/direction_finding_central/src/main.c:88]:
```suggestion
		(void)memcpy(&u64, data->data, data->data_len);
```

File: samples/bluetooth/direction_finding_central/src/main.c
Comment (Thalley) [inline samples/bluetooth/direction_finding_central/src/main.c:84]:
```suggestion
		if (data->data_len > sizeof(u64)) {
```

File: samples/bluetooth/direction_finding_central/src/main.c
Comment (Thalley) [inline samples/bluetooth/direction_finding_central/src/main.c:256]: Consider using the new static CONN_CB_DEFINE (or whatever it is called), as I think that is possibly the suggested way of doing that now

File: samples/bluetooth/direction_finding_central/src/main.c
Comment (Thalley) [inline samples/bluetooth/direction_finding_central/src/main.c:264]:
```suggestion
	err = bt_enable(NULL);
	if (err) {
```

File: samples/bluetooth/direction_finding_central/src/main.c
Comment (Thalley) [inline samples/bluetooth/direction_finding_central/src/main.c:277]: Why is this necessary?

**Review by Thalley — CHANGES_REQUESTED** [review]
Few minor comments, otherwise LGTM

**Review by mbolivar-nordic — APPROVED** [review]
devicetree bits LGTM, thanks

File: samples/bluetooth/direction_finding_central/boards/nrf52833dk_nrf52820.conf
Comment (ppryga) [reply samples/bluetooth/direction_finding_central/boards/nrf52833dk_nrf52820.conf]: Empty line added

File: samples/bluetooth/direction_finding_central/src/main.c
Comment (ppryga) [reply samples/bluetooth/direction_finding_central/src/main.c]: ah yes, leftover from tests

File: samples/bluetooth/direction_finding_central/src/main.c
Comment (ppryga) [reply samples/bluetooth/direction_finding_central/src/main.c:72]: I can do it but in separate PR. Then I'll update all samples that use similar code.

File: samples/bluetooth/direction_finding_central/src/main.c
Comment (ppryga) [reply samples/bluetooth/direction_finding_central/src/main.c]: Added cast. BTW. is `-Wunused-result` enabled in build system?

File: samples/bluetooth/direction_finding_central/src/main.c
Comment (ppryga) [reply samples/bluetooth/direction_finding_central/src/main.c]: Changed

File: samples/bluetooth/direction_finding_central/src/main.c
Comment (ppryga) [reply samples/bluetooth/direction_finding_central/src/main.c]: Thanks I didn't know about `BT_CONN_CD_DEFINE`.

File: samples/bluetooth/direction_finding_central/src/main.c
Comment (ppryga) [reply samples/bluetooth/direction_finding_central/src/main.c]: Removed empty line

File: samples/bluetooth/direction_finding_central/src/main.c
Comment (ppryga) [reply samples/bluetooth/direction_finding_central/src/main.c]: Leftover from tests, removed.

File: samples/bluetooth/direction_finding_central/src/main.c
Comment (Thalley) [reply samples/bluetooth/direction_finding_central/src/main.c]: I don't think so; we have many calls without explicit return handling.

**Review by Thalley — APPROVED** [review]

File: samples/bluetooth/direction_finding_central/src/main.c
Comment (alwa-nordic) [inline samples/bluetooth/direction_finding_central/src/main.c:191]: Is this supposed to be here? I don't see any corresponding push.

Comment (ppryga) [reply samples/bluetooth/direction_finding_central/src/main.c]: Nope, thanks for spotting that!

**Review by kruithofa — APPROVED** [review]

**Review by jhedberg — APPROVED** [review]
Looks fine, but some CI checks are failing..

**Comment by ppryga** [comment]
> Looks fine, but some CI checks are failing..

Yes, I'm waiting for someone to merge: #41905. Then I'll rebase this and other DF sample PR and push it to pass CI.

**Comment by ppryga** [comment]
Fixed build issue.

**Comment by ppryga** [comment]
@jhedberg @cfriedt @nashif The PR is ready for merge

**Pull request merged**
